rpcserver.go

```diff
@@ -3902,17 +3904,20 @@ func createChannelConstraint(
 	}
 }
 
+// isPrivate evaluates the ChannelFlags of the db channel to determine if the
+// channel is private or not.
+func isPrivate(dbChannel *channeldb.OpenChannel) bool {
+	return dbChannel.ChannelFlags&lnwire.FFAnnounceChannel != 1
```

Comment:
We should add a nil check here since in the case above, the `historical` channel can actually be nil. So probably just return `false` if it's `nil`.

Reply:
done.